### ISSUE TITLE
`COMException` 800705b4 during `EncureCoreWebView2Async`

### ISSUE THREAD

**Issue by exchange321** [MicrosoftEdge/WebView2Feedback#2400]
**Description**
We have noticed that some users encountered the following error when initializing WebView2
```
System.Runtime.InteropServices.COMException: This operation returned because the timeout period expired. (Exception from HRESULT: 0x800705B4)
  ?, in void Marshal.ThrowExceptionForHRInternal(int errorCode, IntPtr errorInfo)
  ?, in async Task<CoreWebView2Controller> CoreWebView2Environment.CreateCoreWebView2ControllerAsync(IntPtr ParentWindow)
  ?, in async Task WebView2.EnsureCoreWebView2Async(CoreWebView2Environment environment)+Init(?)
  ?, in async void MainViewModel.Initialize(WebView2 webView2)
  ?, in void AsyncMethodBuilderCore.ThrowAsync(Exception exception, SynchronizationContext targetContext)+(object state) => { }
  ?, in object ExceptionWrapper.InternalRealCall(Delegate callback, object args, int numArgs)
  ?, in object ExceptionWrapper.TryCatchWhen(object source, Delegate callback, object args, int numArgs, Delegate catchHandler)
```

**Version**
SDK: 1.0.1020.30
Runtime: >=95.0.1020.30
Framework: WPF (.NET Framework 4.8)
OS: >=10.0.22000.0

**Repro Steps**
We're unable to reproduce the issue but we can observe the occurrences of ^ from our logs.

**Comment by ShaunLoganOracle** [comment]
Same as/similar to #2054 ?

**Comment by victorthoang** [comment]
Hello,

I noticed at the end of of bug #2054, one of our principal devs made a strong comment:

The creation failure of timeout is due to multiple apps using the same user data folder and they use different options for WebView2 environment. This is not allowed. So, different apps should use different user data folder. FWIW, it should fail with HRESULT_FROM_WIN32(ERROR_INVALID_STATE) instead of timeout. That is a regression in recent releases and will be fixed in future releases.
For ClickOnce apps, ApplicationDeployment.CurrentDeployment.DataDirectory should be a good value to use for WebView2 user data folder.

Is that helpful at all?

**Comment by exchange321** [comment]
Hi, I'm quite new to Windows app development so please let me know if I'm wrong about something.

> The creation failure of timeout is due to multiple apps using the same user data folder and they use different options for WebView2 environment.

The app is made a singleton so this is likely not the case, and the options shouldn't be different since they are hardcoded, not customizable by the user.

> For ClickOnce apps, ApplicationDeployment.CurrentDeployment.DataDirectory should be a good value to use for WebView2 user data folder.

The app is only published to and available in MS Store. Which value will be the most appropriate in this case? The code which we initialize WebView2:
```cs
CoreWebView2Environment environment = await CoreWebView2Environment.CreateAsync(userDataFolder: Program.AppFolder.Path);

await WebView2.EnsureCoreWebView2Async(environment: environment);
```

**Comment by champnic** [comment]
Besides multiple apps, is there any way your app could have 2 WebView2's at the same time? Or tearing one down and creating another? The key to the issue is usually that the environment and command line options need to be the same when sharing a user data folder.

Does the issue go away if you specify a new user data folder each time?

**Comment by exchange321** [comment]
> Besides multiple apps, is there any way your app could have 2 WebView2's at the same time? Or tearing one down and creating another?
No, WebView2 is only initiated when the app launches. There is no retry logic.

> Does the issue go away if you specify a new user data folder each time?
Not sure, we ain't able to reproduce it locally but only noticed it in our logs. Would doing ^ reset the browser every time it launches (e.g., local storage)?

**Comment by champnic** [comment]
Yes - a new user data folder would reset cache/history/local storage each time. It's not recommended to create a new one each time, but as a test would help us figure out if it resolved the issue. Though if you don't have a local repro it's probably not a helpful test regardless.

@LiangTheDev Any other thoughts on what could cause a timeout here?

**Comment by LiangTheDev** [comment]
Timeout could happen if the machine is really slow/busy, there is an existing webview with in compatible configurations, or an existing webview that is shutting down. An existing WebView could be one created in previous window and we are navigating to a new window that also has a WebView.
Try to capture the exception so that the app doesn't crash, and log info that could help figure out what might be the reason for the timeout, like how long does it take for the app to do other initialization, how long has the up been running (so that we know whether it is possible that we are navigating to another app window), an other app state so that we could know how often this timeout happen and whether there is any pattern.
FWIW, ApplicationData.Current.LocalFolder.Path is a good user data folder value for Store app. It is actually the default for Store app that we use.

**Comment by exchange321** [comment]
@LiangTheDev thanks for looking into this. I'll add some logging and see if I can get more info

**Comment by sg-cd** [comment]
We ran into this issue in our product once when the Edge webview processes were orphaned - the parent process was killed, but the Edge webview processes remained. On subsequent restarts, we kept seeing this crash, till we killed the Edge webview processes manually.
Any pointers on what could cause the Edge webview processes to become orphaned?

**Comment by LiangTheDev** [comment]
The Edge WebView process becoming orphaned is not expected. Do you know what is the version of Edge WebView2 Runtime?

Combined with the fact that subsequent WebView creation timing out, it indicates that there is some deadlock/hang during Edge WebView process shutting down. If we could create a dump file of the orphaned WebView process, that would help a lot in identifying what the problem is. To create a dump file, you can use [Windows debugging tools](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/debugger-download-tools) to attach to the process and then use .dump command. See https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/-dump--create-dump-file- for more info.

**Comment by sg-cd** [comment]
Edge WebView2 Runtime version - 104.0.1293.54
We are unable to reproduce the issue now, so unable to create the dump file.
